Page rendering error for missing plugin

**Describe the bug**
A plugin without a description field causes a page render error.

**My Rundeck detail**

* Rundeck version: 3.1.0-snapshot
* install type: launcher
* OS Name/version: MacOs
* DB Type/version: H2

**To Reproduce**
Steps to reproduce the behavior:
1. Deploy a plugin that does not have a description property.
2. Click on job view page that uses that step plugin
3. See error

**Expected behavior**
The page rendering should tolerate plugins that do not have a description property.

**Screenshots**

![image](https://user-images.githubusercontent.com/275924/57181764-c3058700-6e4c-11e9-9c25-85722aad4846.png)

The following is written to the service.log
```
Caused by: org.grails.taglib.GrailsTagException: [views/execution/_wfItemView.gsp:77] Error executing tag <stepplugin:display>: Cannot get property 'description' on null object
	at org.grails.gsp.GroovyPage.throwRootCause(GroovyPage.java:473)
	at org.grails.gsp.GroovyPage.invokeTag(GroovyPage.java:415)
	at gsp_rundeck_execution_wfItemView_gsp.run(gsp_rundeck_execution_wfItemView_gsp.groovy:129)
	at org.grails.gsp.GroovyPageWritable.doWriteTo(GroovyPageWritable.java:162)
	at org.grails.gsp.GroovyPageWritable.writeTo(GroovyPageWritable.java:82)
	at org.grails.web.gsp.GroovyPagesTemplateRenderer.makeTemplate(GroovyPagesTemplateRenderer.java:232)
	at org.grails.web.gsp.GroovyPagesTemplateRenderer.render(GroovyPagesTemplateRenderer.java:113)
	at org.grails.plugins.web.taglib.RenderTagLib$_closure7.doCall(RenderTagLib.groovy:367)
	at org.grails.gsp.GroovyPage.invokeTagLibClosure(GroovyPage.java:446)
	at org.grails.gsp.GroovyPage.invokeTag(GroovyPage.java:364)
	... 139 common frames omitted
Caused by: java.lang.NullPointerException: Cannot get property 'description' on null object
	at rundeck.services.FrameworkService.getStepPluginDescription(FrameworkService.groovy:744)
	at rundeck.services.FrameworkService.getPluginDescriptionForItem(FrameworkService.groovy:720)
	at rundeck.PluginTagLib$_closure1.doCall(PluginTagLib.groovy:32)
	at org.grails.gsp.GroovyPage.invokeTagLibClosure(GroovyPage.java:446)
	at org.grails.gsp.GroovyPage.invokeTag(GroovyPage.java:364)
	... 147 common frames omitted
```
**Desktop (please complete the following information):**
 - OS: MacOs
 - Browser Firefox
 - Version: newest

--- a/rundeckapp/grails-app/services/rundeck/services/FrameworkService.groovy
+++ b/rundeckapp/grails-app/services/rundeck/services/FrameworkService.groovy
@@ -790,7 +790,7 @@ class FrameworkService implements ApplicationContextAware, AuthContextProcessor,
      */
     def Description getNodeStepPluginDescription(String type) throws MissingProviderException {
         final described = pluginService.getPluginDescriptor(type, rundeckFramework.getNodeStepExecutorService())
-        described.description
+        described?.description
     }
     /**
      * Return step plugin description of a certain type
@@ -800,7 +800,7 @@ class FrameworkService implements ApplicationContextAware, AuthContextProcessor,
      */
     def Description getStepPluginDescription(String type) throws MissingProviderException{
         final described = pluginService.getPluginDescriptor(type, rundeckFramework.getStepExecutionService())
-        described.description
+        described?.description
     }
 
     /**

--- a/rundeckapp/src/test/groovy/rundeck/services/FrameworkServiceSpec.groovy
+++ b/rundeckapp/src/test/groovy/rundeck/services/FrameworkServiceSpec.groovy
@@ -25,7 +25,10 @@ import com.dtolabs.rundeck.core.common.IRundeckProjectConfig
 import com.dtolabs.rundeck.core.common.NodeEntryImpl
 import com.dtolabs.rundeck.core.common.ProjectManager
 import com.dtolabs.rundeck.core.common.PropertyRetriever
+import com.dtolabs.rundeck.core.execution.workflow.steps.StepExecutionService
+import com.dtolabs.rundeck.core.execution.workflow.steps.node.NodeStepExecutionService
 import com.dtolabs.rundeck.core.plugins.DescribedPlugin
+import com.dtolabs.rundeck.core.plugins.PluggableProviderService
 import com.dtolabs.rundeck.core.plugins.configuration.Description
 import com.dtolabs.rundeck.core.plugins.configuration.DynamicProperties
 import com.dtolabs.rundeck.core.plugins.configuration.Property
@@ -35,6 +38,7 @@ import com.dtolabs.rundeck.plugins.util.PropertyBuilder
 import grails.test.mixin.TestFor
 import org.rundeck.app.spi.Services
 import org.rundeck.core.projects.ProjectConfigurable
+import rundeck.PluginStep
 import spock.lang.Specification
 import spock.lang.Unroll
 
@@ -451,5 +455,36 @@ class FrameworkServiceSpec extends Specification {
             }
             pluga == ['a', 'b', 'c']
             plugb == []
+    }
+
+    @Unroll
+    def "get plugin description for step item nodeStep #nodeStep description #descriptor"() {
+        given:
+            service.rundeckFramework = Mock(Framework)
+            service.pluginService = Mock(PluginService)
+            def step = new PluginStep(type: 'atype', nodeStep: nodeStep)
+            def nodeStepService = Mock(NodeStepExecutionService)
+            def wfStepService = Mock(StepExecutionService)
+            def desc = DescriptionBuilder.builder().name('atype').build()
+        when:
+            def result = service.getPluginDescriptionForItem(step)
+        then:
+            if (nodeStep) {
+                1 * service.rundeckFramework.getNodeStepExecutorService() >> nodeStepService
+                1 * service.pluginService.getPluginDescriptor('atype', nodeStepService) >>
+                (descriptor ? new DescribedPlugin(_, desc, 'atype') : null)
+            } else {
+                1 * service.rundeckFramework.getStepExecutionService() >> wfStepService
+                1 * service.pluginService.getPluginDescriptor('atype', wfStepService) >>
+                (descriptor ? new DescribedPlugin(_, desc, 'atype') : null)
+            }
+            result == (descriptor ? desc : null)
+
+        where:
+            nodeStep | descriptor
+            true     | true
+            true     | false
+            false    | true
+            false    | false
     }
 }

